src/ProfileEndpoints.jl

```diff
@@ -21,11 +21,20 @@ using Serialization: serialize
 #
 #----------------------------------------------------------
 
-function _http_response(binary_data, filename)
-    return HTTP.Response(200, [
-        "Content-Type" => "application/octet-stream"
-        "Content-Disposition" => "attachment; filename=$(repr(filename))"
-    ], body = binary_data)
+function _http_response(binary_data, filename, should_inline_profile = true)
```

Comment:
I think it's surprising that this function, which has a functional name, actually has a secret side-effect, depending on the parameter.

Can you try to fix that? Either with a different name or different code organization, or different param name, etc? You can decide what's best!

Reply:
Changed the name of the function to include the fact that we may write the profile data.